client/lib/media/library-selected-store.js

```diff
@@ -84,9 +84,15 @@ MediaLibrarySelectedStore.getAll = function( siteId ) {
 		return [];
 	}
 
-	return MediaLibrarySelectedStore._media[ siteId ].map( function( itemId ) {
-		return MediaStore.get( siteId, itemId );
-	} );
+	// Avoid keeping invalid items in the selected list.
+	return (
+		MediaLibrarySelectedStore
+			._media[ siteId ]
+			.map( itemId => MediaStore.get( siteId, itemId ) )
+			.filter( ( item ) => {
+				return ( item && item.guid );
```

Comment:
Could you explain what case we're trying to correct using `item.guid` over `itemId` 

Is this correct for media that we're currently uploading? eg, try throttling your internet speed to something slow in dev tools, then drag and drop an image into the post